### PR TITLE
gh-82504: deprecate the MultiLoopChildWatcher

### PR DESCRIPTION
**Pull request by graingert**
* Issue: gh-82504

### CODE REVIEW

File: Misc/NEWS.d/next/Library/2022-07-07-09-04-49.gh-issue-82504.6xtYAz.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+deprecate the ``asyncio.MultiLoopChildWatcher``
```
Comment (kumaraditya303) [inline Misc/NEWS.d/next/Library/2022-07-07-09-04-49.gh-issue-82504.6xtYAz.rst]:
```suggestion
Deprecate :class:`asyncio.MultiLoopChildWatcher` to be removed in Python 3.14.
```

**Review by kumaraditya303 — CHANGES_REQUESTED** [review]
It would be better to add What's New entry about this deprecation.

File: Doc/library/asyncio-policy.rst [hunk 2]
```diff
@@ -232,6 +232,7 @@ implementation used by the asyncio event loop:
    :py:data:`SIGCHLD` is received).
 
    .. versionadded:: 3.8
+   .. deprecated:: 3.12
```
Comment (graingert) [inline Doc/library/asyncio-policy.rst]:
```suggestion
   .. deprecated-remove:: 3.12 3.14
```

File: Doc/library/asyncio-policy.rst [hunk 3]
```diff
@@ -232,6 +232,7 @@ implementation used by the asyncio event loop:
    :py:data:`SIGCHLD` is received).
 
    .. versionadded:: 3.8
+   .. deprecated-remove:: 3.12 3.14
```
Comment (graingert) [inline Doc/library/asyncio-policy.rst]:
```suggestion
   .. deprecated-removed:: 3.12 3.14
```

File: Doc/whatsnew/3.12.rst [hunk 4]
```diff
@@ -171,6 +171,18 @@ APIs:
 * :func:`unittest.getTestCaseNames` (:gh:`50096`)
 * :class:`webbrowser.MacOSX` (:gh:`86421`)
 
+
+Pending Removal in Python 3.14
+==============================
+
+The following modules and APIs have been deprecated in earlier Python releases,
```
Comment (kumaraditya303) [inline Doc/whatsnew/3.12.rst]: Also mention this under `Deprecated` that it is deprecated in  3.12

Comment (graingert) [reply Doc/whatsnew/3.12.rst]: oh this is the first deprecation of the 3.12 release cycle!

File: Doc/whatsnew/3.12.rst [hunk 5]
```diff
@@ -171,6 +178,7 @@ APIs:
 * :func:`unittest.getTestCaseNames` (:gh:`50096`)
 * :class:`webbrowser.MacOSX` (:gh:`86421`)
 
+
```
Comment (graingert) [inline Doc/whatsnew/3.12.rst]:
```suggestion
```

File: Doc/whatsnew/3.12.rst (hunk 4 above)
Comment (graingert) [reply Doc/whatsnew/3.12.rst]: @kumaraditya303 I just realized it's not correct for this notice to go in `Pending Removal in Python 3.14`, it's supposed to be a reminder of previous deprecations, so I moved it to 'Deprecations'

Comment (AlexWaygood) [reply Doc/whatsnew/3.12.rst]: > oh this is the first deprecation of the 3.12 release cycle!

Nah, we already deprecated `typing.Sized` and `typing.Hashable` (in favour of `collections.abc.Sized` and `collections.abc.Hashable`), we just didn't set a removal date for those deprecations ;)

Comment (graingert) [reply Doc/whatsnew/3.12.rst]: bah you just got merged before me :p

**Review by kumaraditya303 — APPROVED** [review]
LGTM

**Comment by graingert** [comment]
thanks for the review @kumaraditya303 !

**Comment by kumaraditya303** [comment]
@gvanrossum Would you like to review this? This deprecates `MultiLoopChildWatcher`  which has race conditions so it is not even tested on CI currently, skipped on CI as of https://github.com/python/cpython/pull/26542.

**Comment by gvanrossum** [comment]
I'm sorry, I don't feel comfortable reviewing this. Any core dev CC'ed on the issue would be more qualified than me.

**Comment by graingert** [comment]
@kumaraditya303 can you request @njsmith on this?

**Comment by kumaraditya303** [comment]
Superseded by https://github.com/python/cpython/pull/98024

**Pull request closed without merging**